### PR TITLE
Validate empty filter condition

### PR DESCRIPTION
**Pull request by agourlay**
context: https://github.com/qdrant/qdrant/issues/2299

This PR adds validation to filter condition so that query where only the key is defined are now considered malformed right away.

### CODE REVIEW

File: openapi/tests/openapi_integration/test_recommend.py [hunk 1]
```diff
@@ -231,10 +231,8 @@ def get_points(ids: list):
 
 
 def test_raw_vectors():
-    points = get_points([1,2,3,4,5,6,7,8])
-    
-    print(points[0])
```
Comment (agourlay) [inline openapi/tests/openapi_integration/test_recommend.py:236]: removed noisy print :broom:

**Review by timvisee — CHANGES_REQUESTED** [review]
Thanks for picking this up!

In two new commits I also enabled `Filter` validation on other request objects, not just `search`. I also made sure this propagates through nested filters.

I do think we should also implement this for gRPC. Due to code generation it is implemented a bit weird, but the validation definitions are [here](https://github.com/qdrant/qdrant/blob/c3657fd9c733943ee27ad01feb4c48cb2397ccef/lib/api/build.rs#L84-L235). Mind to give that a shot?

**Comment by agourlay** [comment]
@timvisee Thanks a lot, I am trying to add the missing bits for gRPC :+1:

**Comment by agourlay** [comment]
@timvisee added gRPC validation in https://github.com/qdrant/qdrant/pull/2745/commits/2f58769f18f303cde49881ae7163aa8cd2a0f02b

Sadly there is duplication between gRPC & REST but the underlying types are different.

WDYT?

File: lib/api/src/grpc/validate.rs [hunk 2]
```diff
@@ -116,6 +116,42 @@ impl Validate for crate::grpc::qdrant::quantization_config_diff::Quantization {
...
+            ConditionOneOf::IsEmpty(_) => Ok(()),
+            ConditionOneOf::HasId(_) => Ok(()),
+            ConditionOneOf::IsNull(_) => Ok(()),
+        }
+    }
+}
+
+impl Validate for crate::grpc::qdrant::FieldCondition {
+    fn validate(&self) -> Result<(), ValidationErrors> {
+        // TODO && self.geo_polygon.is_none()
```
Comment (agourlay) [inline lib/api/src/grpc/validate.rs:135]: this must be added once https://github.com/qdrant/qdrant/pull/2315 lands
@zzzz-vincent

**Review by timvisee — APPROVED** [review]

File: lib/api/src/grpc/validate.rs (hunk 2 above)
Comment (zzzz-vincent) [reply lib/api/src/grpc/validate.rs:135]: ack. Thanks for letting me know.

**Review by coszio — APPROVED** [review]
Simple and effective ✅

I only made a nit edit for exhausting the validation on `Condition` enum. It should make the compiler complain if we add more variants.

**Pull request merged**